controller/loginController.js

```diff
@@ -10,7 +13,46 @@ const login = async (req, res) => {
             return res.status(400).json({ error: 'Username and password are required' });
          }
         const user = await getUserCredentials(username, password);
-        if (user.length === 0) {
+        if (!user || user.length === 0) {
+            return res.status(401).json({ error: 'Invalid username or password' });
+        }
+
+        const isPasswordValid = await bcrypt.compare(password, user.password);
+        if (!isPasswordValid) {
+            return res.status(401).json({ error: 'Invalid username or password' });
+        }
+       
+
+        if (user.mfa_enabled) {
+            let token = Math.floor(100000 + Math.random() * 900000);
+
+            await addMfaToken(user.user_id, token);
+
+            await sendEmail(user, token);
+            return res.status(200).json({ message: 'An MFA Token has been sent to your email address', token: token });
```

Comment:
Probably don't want to send the token itself back otherwise this just bypasses the MFA altogether.

Might be worth using a 202 ACCEPTED status code here as well so it's easy for front end to tell when the login has passed or is going through MFA